run.py

```diff
@@ -10,52 +9,27 @@
 import sqlite3
 import sys
 import time
+import psycopg2
 
 from balloon import *
 from telemetry import *
 
+def to_list(tuple):
+    return list(tuple)
+
 def getspots (nrspots):
-#    logging.info("Fetching...")
-    wiki = "http://wsprnet.org/olddb?mode=html&band=all&limit=" + str(nrspots) + "&findcall=&findreporter=&sort=spotnum"
+    # The connection credentials are public, i.e. https://inductivestep.github.io/WSPR-analysis/
     try:
-        page = requests.get(wiki)
-    except requests.exceptions.RequestException as e:
-        print("ERROR",e)
-        return []
-
-#    logging.info(page.status)
-#    logging.info(page.data)
-
-    soup = BeautifulSoup(page.content, 'html.parser')
-
-    data = []
-    table = soup.find_all('table')[2]
-    #print("TABLE:",table)
-
-    rows = table.findAll('tr')
-    for row in rows:
-        cols = row.find_all('td')
-        cols = [ele.text.strip() for ele in cols]
-        data.append([ele for ele in cols if ele]) # Get rid of empty values
-
-    # Strip empty rows
-    newspots = [ele for ele in data if ele] 
-
-    # Strip redundant columns Watt & miles and translate/filter data
-    for row in newspots:
-        #print(row)
-        row[0] = datetime.datetime.strptime(row[0], '%Y-%m-%d %H:%M')
-        row[6] = int(row[6].replace('+',''))
-
-        del row[12]
-        del row[11]
-        del row[7]
-
-    # Reverse the sorting order of time to get new spots firsts
-    newspots.reverse()
-
-    return newspots
-
+        conn = psycopg2.connect("host=logs2.wsprdaemon.org dbname=wsprnet user=wdread password=JTWSPR2008")
+    except psycopg2.OperationalError as err:
+        logging.error("PostgreSQL connect error: " + err)
+        conn = None
```

Comment:
Not too familiar with Python, but isn't it necessary to return in the exception handler, so the `conn.cursor()` on the next line would not be executed?